common/tools/dev-tool/src/commands/admin/migrate-package.ts

```diff
@@ -266,7 +266,7 @@ function setScriptsSection(scripts: PackageJson["scripts"]): void {
   scripts["build"] = "npm run clean && dev-tool run build-package && dev-tool run extract-api";
 
   scripts["unit-test:browser"] =
-    "npm run clean && dev-tool run build-package && dev-tool run build-test && dev-tool run test:vitest --no-test-proxy --browser";
+    "npm run clean && dev-tool run build-package && dev-tool run build-test && dev-tool run test:vitest --browser";
```

Comment:
Would we assume the test-proxy is required for unit tests as well as for integration tests?

Reply:
I think the `no-test-proxy` flag should be reserved for packages that don’t use recordings at all, but I could be overlooking something.

Currently, the `TEST_MODE` environment variable seems to guide whether we use the proxy or not:

- playback: true
- recording: true
- live: false

However, the challenge I see is the distinction between `unit-test` and `integration-test`. This split seems to come from when we used to run `unit-test` from source code and `integration-test` from the distribution (JavaScript). But now that we’ve consolidated everything under Vitest, it raises the question: should `integration-test` only be used for live tests now? Or should we just rely on `TEST_MODE` entirely?

Another option could be to remove the `integration-test` script altogether, as running `TEST_MODE=live rushx unit-test` would achieve the same result as the current `integration-test`. This might be something worth discussing with the team, as the purpose behind keeping both scripts isn't entirely clear to me anymore.

idk, what do folks think?

Reply:
I'll bring this up in the team meeting tomorrow and see what folks think. Meanwhile going to resolve the comment 👍

Reply:
messaging (amqp) tests don't need test-proxy, although running them probably won't hurt anything

> However, the challenge I see is the distinction between unit-test and integration-test. This split seems to come from when we used to run unit-test from source code and integration-test from the distribution (JavaScript). 

I think there's difference between unit-test and integration-tests, it is just that most of our packages treat them as the same with playback.  But some package might choose difference set of test files between the two.  Also some tests may not be suitable for playback (uploading huge files for example).